File: packages/shared/src/constants/keywords.ts
import keywords from "./keywords.json";

// 通知解析场景（apps/mobile/lib/auto-bookkeeping/parser.ts）：
// 精简表，只含明确的指示动词。窄匹配避免把"工资代发提醒"之类的
// 银行推广文本误识别为到账收入。
export const NOTIFICATION_INCOME_KEYWORDS: readonly string[] =
  keywords.notification.income;
export const NOTIFICATION_EXPENSE_KEYWORDS: readonly string[] =
  keywords.notification.expense;

export const NOTIFICATION_INCOME_REGEX = new RegExp(
  NOTIFICATION_INCOME_KEYWORDS.join("|"),
);
export const NOTIFICATION_EXPENSE_REGEX = new RegExp(
  NOTIFICATION_EXPENSE_KEYWORDS.join("|"),
);

// LLM 语义判断场景（apps/backend/services/silicon.py）：
// 宽表，涵盖名词短语 + 动词 + 不同行业用词，用于提示 LLM 如何
// 区分自然语言里的收入 / 支出意图。
export const SEMANTIC_INCOME_KEYWORDS: readonly string[] =
  keywords.semantic.income;
export const SEMANTIC_EXPENSE_KEYWORDS: readonly string[] =
  keywords.semantic.expense;

File: apps/backend/constants/keywords.py
"""收入 / 支出语义关键词表。

单一事实源位于 packages/shared/src/constants/keywords.json，
前端（apps/mobile）通过 @coco/shared 导入对应常量，
后端在模块加载时读取同一份 JSON，以保持前后端语义一致。
"""

import json
from pathlib import Path

_REL = Path("packages/shared/src/constants/keywords.json")


def _find_keywords_json() -> Path:
    for parent in Path(__file__).resolve().parents:
        candidate = parent / _REL
        if candidate.is_file():
            return candidate
    raise FileNotFoundError(f"未找到 {_REL}（已向上检索至文件系统根）")


_data = json.loads(_find_keywords_json().read_text(encoding="utf-8"))

NOTIFICATION_INCOME_KEYWORDS: list[str] = _data["notification"]["income"]
NOTIFICATION_EXPENSE_KEYWORDS: list[str] = _data["notification"]["expense"]
SEMANTIC_INCOME_KEYWORDS: list[str] = _data["semantic"]["income"]
SEMANTIC_EXPENSE_KEYWORDS: list[str] = _data["semantic"]["expense"]
